californium-core/src/main/java/org/eclipse/californium/core/network/CorrelationStrategyFactory.java

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - introduce CorrelationStrategy
+ *                                      (fix GitHub issue #104)
+ ******************************************************************************/
+package org.eclipse.californium.core.network;
+
+import org.eclipse.californium.core.network.config.NetworkConfig;
+import org.eclipse.californium.elements.CorrelationStrategy;
+import org.eclipse.californium.elements.RelaxedCorrelationStrategy;
+import org.eclipse.californium.elements.StrictlyCorrelationStrategy;
+
+/**
+ * Factory for correlation strategy.
+ */
+public abstract class CorrelationStrategyFactory {
```

Comment:
I am not a big fan of using such *factory* classes instead of passing in an instance of the interface to be created. However, I see your point, @boaks, regarding the somewhat *inflexible* process of creating and configuring a `CoapEndpoint` that can be used by leshan (this is addressed by the `improve_builder` branch already which provides means to set `Endpoint`s explicitly).
So, AFAIC it would be preferrable to allow for client code to pass in an instance of `CorrelationStrategy` to the `CoapEndpoint` constructor and, if the instance is `null` use this (private) factory for creating an instance based on the configuration property. However, I would rather not expose this factory as public API to be extended and used by client code.